File: examples/octave/TwymanGreen.m
% LightPipes for Octave Optical Toolbox
% Twyman Green interferometer.
% The Twyman green interferometer consists of a beamsplitter
% and two mirrors. One of the mirrors
% is perfect, the other can be investigated for aberrations. 
%
% F.A. van Goor, March 1998.

clear;

m=1;
cm=1e-2*m;
mm=1e-3*m;
nm=1e-9*m;

size=30*mm;
lambda=500*nm;
R=5*mm;
N=250;
z1=50*cm;
z2=40*cm;
z3=40*cm;
z4=100*cm;
RBS=0.3;
nz=3;
mz=1;
Rz=0.005;
Az=25;

F=LPBegin(size,lambda,N); F=LPCircAperture(R,0,0,F);
F=LPForvard(z1,F);
F1 = F2 = F; F1.F *= RBS; F2.F *= 1-RBS;
F1=LPForvard(2*z2,F1); F2=LPForvard(z3,F2);
F2=LPZernike(nz,mz,Rz,Az,F2);
F2=LPForvard(z3,F2);
F1.F *= RBS; F2.F *= 1-RBS;
F=LPBeamMix(F1,F2);
F=LPForvard(z4,F); F=LPInterpol(0.012,250,0,0,0,1,F);
Int=F.F .* conj(F.F);

figure(1);
mymesh(Int)
title('Twyman Green interferometer with Zernike aberration');

File: examples/python/TwymanGreen.py
#!/usr/bin/env python
"""
LightPipes for Python Optical Toolbox
Twyman Green interferometer.
The Twyman green interferometer consists of a beamsplitter
and two mirrors. One of the mirrors
is perfect, the other can be investigated for aberrations. 

F.A. van Goor, March 1998.
"""

import common
from lightpipes import Field
from pylab import *

m=1
cm=1e-2*m
mm=1e-3*m
nm=1e-9*m

wavelength  = 500*nm
size        = 30*mm
R           = 5*mm
N           = 250
z1          = 50*cm
z2          = 40*cm
z3          = 40*cm
z4          = 100*cm
RBS         = 0.3
nz          = 3
mz          = 1
Rz          = 0.005
Az          = 25

F1 = Field(N,size,wavelength)
F1.circular_aperture(R, 0, 0)
F1.forvard(z1)

F2 = F1.copy()

F1 *= RBS;          F2 *= 1-RBS
F1.forvard(2*z2);   F2.forvard(z3)
F1;                 F2.zernike(nz, mz, Rz, Az)
F1;                 F2.forvard(z3)
F1 *= RBS;          F2 *= 1-RBS

F = F1 + F2

F.forvard(z4)
F.interpolate(0.012, 250, 0, 0, 0, 1)

imshow( abs(F.value)**2 ) # plot intensity
title('Twyman Green interferometer with Zernike aberration')
show()
